chef/cookbooks/utils/libraries/service.rb

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright 2017, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# As we want to override the restart method to either call it or skip it,
+# we just monkey patch the Chef::Provider::Service to override the action_restart
+# which is the common method for all the children Service classes
+class Chef
+  class Provider
+    class Service < Chef::Provider
+      def action_restart
+        restart_manager = ServiceRestart::RestartManager.new(cookbook_name, node)
+
+        if restart_manager.disallow_restart?
+          Chef::Log.info("Disallowing restart for #{@new_resource.name} due to flag")
+          restart_manager.register_restart_request(@new_resource.name.to_sym, false)
```

Comment:
I noticed that we have moved away from creating the RestartManager with the service name, and instead we are adding it as a parameter for each of the methods - why didn't we go with specifying the service name for the constructor?

Reply:
Im not sure when did that happen, I dont think I changed that, but I can see that on the crowbar-pacemaker the signature for the RestartManager is different :/

Maybe I magled it somehow, let me re-fix that :(